core/domain/customization_args_util.py

```diff
@@ -118,8 +120,9 @@ def validate_customization_args_and_values(
                 schema_utils.normalize_against_schema(
                     customization_args[ca_spec.name]['value'],
                     ca_spec.schema))
-        except Exception:
+        except Exception as e:
             # TODO(sll): Raise an actual exception here if parameters are
             # not involved (If they are, can we get sample values for the
             # state context parameters?).
-            pass
+            if fail_on_validation_errors:
+                raise utils.ValidationError(e)
```

Comment:
Is there a need to preserve the stacktrace? I'm not sure.

Reply:
I don't think so as of now, if we have the exact issues in the customization args, we can find a way to fix them through migration and the way to prevent them in future would be to enable the `fail_on_validation_errors` flag.

Reply:
I still think the stack trace would be helpful (so we know where in the validation the error came from).

To preserve it all we need to do is use a bare raise, like so:
```suggestion
                raise
```

Reply:
Hmm, but wouldn't that be too long to show in the output of the job?

Reply:
Just as an FYI, this method is also used in state_domain so the error log will be presented to the client-side, so I think we should not add stack trace in the error message. (If required you can use logger, maybe?)

Reply:
I am planning to do a run just with these errors and see if anything extra is required. I don't think we would need more info, as for html validation run by @bansalnitish, just the error messages for what is wrong with html were sufficient to fix the issues. @brianrodri thoughts?